tests/ci_visibility/test_ci_visibility.py

```diff
@@ -88,3 +89,10 @@ def test_repository_name_not_extracted_warning():
         extracted_repository_name = _extract_repository_name_from_url(repository_url)
         assert extracted_repository_name == repository_url
     mock_log.warning.assert_called_once_with("Repository name cannot be parsed from repository_url: %s", repository_url)
+
+
+def test_civisibilitywriter_agentless_url():
+    with override_env(dict(DD_API_KEY="foobar.baz")):
```

Comment:
Can we add testing both cases where we set the flag via environment variable and via the global config?
i.e. 
```python
with override_env(dict(DD_API_KEY="foobar.baz", DD_CIVISIBILITY_AGENTLESS_URL="https://foo.bar")):
    dummy_writer =  DummyCIVisibilityWriter()
    assert dummy_writer.intake_url == "https://foo.bar"
```